Parse/Internal/LocalDataStore/SQLite/PFSQLiteStatement.m

```diff
@@ -11,13 +11,23 @@
 
 #import <sqlite3.h>
 
+#import "PFThreadsafety.h"
+
+// PFThreadsafetySafeDispatchSync, but with a return type.
+#define thread_safe_sync(queue, block) ({                          \
+    __block typeof((block())) result;                              \
+    PFThreadsafetySafeDispatchSync(queue, ^{ result = block(); }); \
+    result;                                                        \
+})
+
 @implementation PFSQLiteStatement
 
-- (instancetype)initWithStatement:(sqlite3_stmt *)stmt {
+- (instancetype)initWithStatement:(sqlite3_stmt *)stmt queue:(dispatch_queue_t)databaseQueue {
     self = [super init];
     if (!stmt || !self) return nil;
```

Comment:
We really need to clean this up at some point... `if (!stmt)` sounds pretty damn bad...